fedimint-core/src/api.rs

```diff
@@ -500,6 +557,98 @@ pub trait IGlobalFederationApi: IRawFederationApi {
         timeout: Duration,
         num_responses_required: Option<usize>,
     ) -> FederationResult<ApiVersionSet>;
+
+    /// Sets the password used to decrypt the configs and authenticate
+    ///
+    /// Must be called first before any other calls to the API
+    async fn set_password(&self, auth: ApiAuth) -> FederationResult<()>;
```

Comment:
> Must be called first before any other calls to the API

This doesn't make a lot of sense if read in the context of the normal client API. Maybe the admin API should be defined as an extension trait just like module APIs are? I think that would make it easier to understand the separate concepts of "normal API" and "admin API"

Reply:
I just moved it.